lucene/core/src/java/org/apache/lucene/codecs/blocktree/BlockTreeTermsWriter.java

```diff
@@ -1060,36 +1052,35 @@ public void close() throws IOException {
       return;
     }
     closed = true;
-    
+
+    final String metaName = IndexFileNames.segmentFileName(state.segmentInfo.name, state.segmentSuffix, BlockTreeTermsReader.TERMS_META_EXTENSION);
     boolean success = false;
-    try {
-      
-      final long dirStart = termsOut.getFilePointer();
-      final long indexDirStart = indexOut.getFilePointer();
+    try (IndexOutput metaOut = state.directory.createOutput(metaName, state.context)) {
```

Comment:
why this file is not created at the same time with the indexOut, termOut?

Reply:
That would work too. I like keeping the index output open for as little time as possible when it doesn't make things worse otherwise.